pkg/operator/deploy/deploy.go

```diff
@@ -374,53 +374,57 @@ func (o *operator) IsReady(ctx context.Context) (bool, error) {
 	return true, nil
 }
 
-func checkOperatorDeploymentVersion(ctx context.Context, cli appsv1client.DeploymentInterface, name string, gitCommit string) error {
+func checkOperatorDeploymentVersion(ctx context.Context, cli appsv1client.DeploymentInterface, name string, gitCommit string) (bool, error) {
 	d, err := cli.Get(ctx, name, metav1.GetOptions{})
-	if err != nil {
-		return err
+	switch {
+	case kerrors.IsNotFound(err):
+		return false, nil
+	case err != nil:
+		return false, err
 	}
-
 	if d.Labels["version"] != gitCommit {
-		return errors.New(name + " is not running the desired version: " + gitCommit)
+		return false, nil
 	}
-
-	return nil
+	return true, nil
 }
 
-func checkPodImageVersion(ctx context.Context, cli corev1client.PodInterface, namespace string, gitCommit string) error {
+func checkPodImageVersion(ctx context.Context, cli corev1client.PodInterface, namespace string, gitCommit string) (bool, error) {
```

Comment:
The `namespace` parameter is misleading.  This isn't actually the namespace, but the "role".  Please rename this parameter to `role` as the role of the operator is either a `worker` or `master`.